### ISSUE TITLE
ERROR: Failed building wheel for hdbscan

### ISSUE THREAD

**Issue by Pablo-GDT** [scikit-learn-contrib/hdbscan#428]
When I run the command pip install hdbscan, I get the following error: 

RuntimeError: The current Numpy installation ('C:\\Users\\Pablo GDT\\AppData\\Local\\Temp\\pip-build-env-7165tsmr\\overlay\\Lib\\site-packages\\numpy\\__init__.py') fails to pass a sanity check due to a bug in the windows runtime. See this issue for more information: https://tinyurl.com/y3dm3h86
ERROR: Failed building wheel for hdbscan
Failed to build hdbscan
ERROR: Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly

I have no clue what this means but I've followed threads similar to this where it's suggested to use numpy==1.19.3 which I modified in both my base and venv. I would appreciate any guidance on the issue and thank you in advance!

**Comment by Belval** [comment]
To add to this, there is a bug with Numpy on Windows right now for which the advised fix it to downgrade numpy to 1.19.3. Unfortunately, that does not fix the issue for `hdbscan` which seems to be triggering the exception even if the numpy version is correct.

TLDR: As of right now the package cannot be installed on Windows.

**Comment by hp0404** [comment]
I'm stuck with the same problem -- `ERROR: Could not build wheels for hdbscan which use PEP 517 and cannot be installed directly` -- but I'm using Linux (Ubuntu 20.04.1), Python 3.8.5

**Comment by billfreeman44** [comment]
I ran into this today and [Installing Visual C++ 2015 Build Tools](https://stackoverflow.com/questions/39771592/visual-c-for-python-failed-with-exit-status-2-when-installing-divisi2/55493622#55493622) fixed it for me.

**Comment by zhaoyi1025** [comment]
Hi @Belval， When I run the command ` pip install E:\zy\hdbscan-master`, (I need install the latest codes to avoid the memory mapping issues), I get the following error:
`Processing e:\zy\hdbscan-master
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: joblib>=1.0 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (1.1.0)
Requirement already satisfied: scipy>=1.0 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (1.6.2)
Requirement already satisfied: scikit-learn>=0.20 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (1.0.1)
Requirement already satisfied: numpy>=1.20 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (1.20.3)
Requirement already satisfied: cython>=0.27 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (0.29.14)
Requirement already satisfied: six in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from hdbscan==0.8.27) (1.16.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in c:\users\amws\.conda\envs\top2vec\lib\site-packages (from scikit-learn>=0.20->hdbscan==0.8.27) (3.0.0)
Building wheels for collected packages: hdbscan
  Building wheel for hdbscan (pyproject.toml) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\AMWS\.conda\envs\top2vec\python.exe' 'C:\Users\AMWS\.conda\envs\top2vec\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Users\AMWS\AppData\Local\Temp\tmp0c8o7bdd'
       cwd: E:\zy\hdbscan-master
  Complete output (15 lines):
  running bdist_wheel
  running build
  running build_py
  running build_ext
  skipping 'hdbscan\_hdbscan_tree.c' Cython extension (up-to-date)
  skipping 'hdbscan\_hdbscan_linkage.c' Cython extension (up-to-date)
  skipping 'hdbscan\_hdbscan_boruvka.c' Cython extension (up-to-date)
  skipping 'hdbscan\_hdbscan_reachability.c' Cython extension (up-to-date)
  skipping 'hdbscan\_prediction_utils.c' Cython extension (up-to-date)
  skipping 'hdbscan\dist_metrics.c' Cython extension (up-to-date)
  building 'hdbscan._hdbscan_tree' extension
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\AMWS\.conda\envs\top2vec\include -IC:\Users\AMWS\.conda\envs\top2vec\include -IC:\Users\AMWS\AppData\Local\Temp\pip-build-env-0rf1jlnu\overlay\Lib\site-packages\numpy\core\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30037\include" /Tchdbscan\_hdbscan_tree.c /Fobuild\temp.win-amd64-3.8\Release\hdbscan\_hdbscan_tree.obj
  _hdbscan_tree.c
  C:\Users\AMWS\.conda\envs\top2vec\include\pyconfig.h(59): fatal error C1083: 无法打开包括文件: “io.h”: No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Professional\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for hdbscan
Failed to build hdbscan
ERROR: Could not build wheels for hdbscan, which is required to install pyproject.toml-based projects`
Is there any suggestions about this issue? Thanks in advance.

**Comment by shachafl** [comment]
Same problem running the command: python -m pip install hdbscan
on ubuntu 20.04.5 (WSL 2) and fresh venv of python3.8.10 after upgrading pip setuptools wheel to their latest versions.

To solve I had to install gcc, build-essential & python3.8-dev:
sudo apt install gcc build-essential python3.8-dev
